### ISSUE TITLE
Script Actions to Stash/Unstash party and inventory

### ISSUE THREAD

**Issue by xirsoi** [Tuxemon/Tuxemon#724]
To support the Crawl mechanic, or other story events, we should have actions for temporarily storing the player's party and inventory.

Several options present themselves, each with their own pros and cons:

1) Store them in game variables. Pro is that the script could dictate where it goes, though I'm not sure what good that would do. Con is that we'd have to make sure the Save function would know to serialize monsters and items correctly, as it currently only handles simple values like strings and numbers.

2) Store them in dedicated properties of the Player class. Much the same situation as in game variables, but it would be easier to update the Save function. Con is that we would have no ability to stash multiple parties/backpacks.

3) Simply drop all of it in computer storage. Pro is it's easily the simplest solution with few problems from an engineering perspective. Con is that the UX is not great. The player would have to manually find all their shit and pull it out of storage again. Another Con is that the computer storage mechanics haven't been written yet.

**Comment by bitcraft** [comment]
> Store them in game variables

Save/load should be working just fine.  The simple values in there should be used to create new in-game instances of monsters, items, etc.  So, storing it in game variables seems fine since it will be json text.

> Store them in dedicated properties of the Player class

Minor point, this would be NPC.  I don't want NPCs and "Players" to differ substantially, since it is easier to maintain code that works on both.  I'm not sure that I like this suggestion either though, due to "backpacks", and such thing you mentioned.

> Simply drop all of it in computer storage

Computer storage should be a subset of "multiple inventories".  Ideally, I'd like a medium-sized refactor of inventories by replacing the dictionary with a class that we can serialize and apply attributes to, such as name, size, type of objects, etc.  Then we could develop a class which displays inventories to reuse code.  Way outside the scope of the ticket though.

I like the #1 approach best.  When a crawl starts, the current inventory is serialized then replaced with an empty one.  When the crawl ends, the inventory is dropped again and restored from the saved one.  If there are any deficiencies in the save/load code, they should be fixed.  In the short term, I think this is the best solution if we have momentum on the crawls feature.

**Comment by xirsoi** [comment]
Completely agree with all of that.

I don't know if there's currently momentum on the feature, but I stumbled on the discussion in the forums relatively recently and really liked the idea. I thought at the very least I could introduce a ticket for one of the features required to get it done.

Another thought is that we'll need to provide some way to have the player select monsters (from... somewhere?) to include during the crawl, with restrictions. There are multiple ways this could be done, but it'll require a discussion with the content people to determine what way we think is best to implement.

**Comment by bitcraft** [comment]
Yeah.  I really like the idea too and would love to see it happen.  Monster selection ideally would be done on a new screen to assemble a party.  The existing monsters screen isn't adequate.  I have a bunch of GUI code that I need to get up to date that would work for a party screen/menu.

https://gyazo.com/a3604585ef0160c9ab96a8159fae9745
